src/LibraryManager/Providers/Cdnjs/CdnjsProvider.cs

```diff
@@ -139,6 +122,56 @@ public ILibraryCatalog GetCatalog()
             return LibraryInstallationResult.FromSuccess(desiredState);
         }
 
+        /// <summary>
+        /// Updates file set on the passed in ILibraryInstallationState in case user selected to have all files included
+        /// </summary>
+        /// <param name="desiredState"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        public async Task<ILibraryInstallationResult> UpdateStateAsync(ILibraryInstallationState desiredState, CancellationToken cancellationToken)
+        {
+            try
+            {
+                if (desiredState.Files != null && desiredState.Files.Count > 0)
+                {
+                    return LibraryInstallationResult.FromSuccess(desiredState);
+                }
+
+                var catalog = (CdnjsCatalog)GetCatalog();
+                ILibrary library = await catalog.GetLibraryAsync(desiredState.LibraryId, cancellationToken).ConfigureAwait(false);
+
+                if (library == null)
+                {
+                    throw new InvalidLibraryException(desiredState.LibraryId, Id);
+                }
+
+                await HydrateCacheAsync(library, cancellationToken).ConfigureAwait(false);
+
+                desiredState = new LibraryInstallationState
+                {
+                    ProviderId = Id,
+                    LibraryId = desiredState.LibraryId,
+                    DestinationPath = desiredState.DestinationPath,
+                    Files = library.Files.Keys.ToList(),
+                };
+            }
+            catch (Exception ex) when (ex is InvalidLibraryException || ex.InnerException is InvalidLibraryException)
```

Comment:
Should we remove the try/catch for (InvalidLibraryException || ex.InnerException is InvalidLibraryException)  for the callers of UpdateStateAsync ?